testing/src/test/java/com/hazelcast/samples/testing/samples/junit4/FlakyTest.java

```diff
@@ -0,0 +1,60 @@
+package com.hazelcast.samples.testing.samples.junit4;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import com.hazelcast.test.HazelcastSerialClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.annotation.Repeat;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(HazelcastSerialClassRunner.class)
+public class FlakyTest extends HazelcastTestSupport {
+
+    private static final AtomicInteger run = new AtomicInteger();
+    private static final AtomicInteger counter = new AtomicInteger();
+    private HazelcastInstance member1;
+
+    @Before
+    public void setUp() {
+        member1 = createHazelcastInstance();
+    }
+
+    @After
+    public void tearDown() {
+        if (member1 != null) {
+            member1.shutdown();
+        }
+    }
+
+    @Repeat(5)
+    @Ignore
+    @Test
+    public void testFlakyBehavior() {
+
+        IMap<String, Integer> map = member1.getMap("map");
+        map.put("key", 0);
+
+        // simulate intermittent behavior: succeed only half the time
+        if (System.currentTimeMillis() % 2 == 0) {
```

Comment:
```suggestion
        if (System.nanoTime() % 2 == 0) {
```

Turbo-nit - `currentTime` isn't monotonic which _theoretically_ could alter your stats

Reply:
cool